Added Docs related to excluding namespaces from the Knative Webhook (#2722)
knative/docs: docs/serving/webhook-customizations.md

---
title: "Webhook Customizations"
weight: 70
type: "docs"
---

> 

Reviewer: Add some context information about why a user would want to /  need to do this?

> ## Exclude Namespaces from Webhook

Reviewer: ```suggestion
## Exclude namespaces from the Knative webhook
```

Reviewer: Actually I'd rename the doc 
"Webhook Customizations" -> "Exclude namespaces from the Knative webhook"
and then remove this ## heading entirely, since you're not really customizing, just disabling.

Author: Good idea. I was trying to set this up as a doc that may include other customizations. Trying to figure out if this is better as a single doc or added to something else. I'll update it like it is always going to be a solo doc and use the title you suggested.

Reviewer: Yeah totally know what you mean :slightly_smiling_face: I used to do this too, but I think it's actually better just to keep it specific and then if it needs to move into a bigger "webhooks" section or something later, it can. Thanks so much for making the changes!

> A namespace can use the label `webhooks.knative.dev/exclude` to bypass the Knative Webhook.

Reviewer: ```suggestion
You can configure the label `webhooks.knative.dev/exclude` to allow namespaces to bypass the Knative webhook.
```

``` yaml
apiVersion: v1
kind: Namespace
metadata:
  name: knative-dev
  labels: 
    webhooks.knative.dev/exclude: "true"
```

> By default the webhook will examine resources that are created/read/updated/deleted. This includes system namespaces which can cause issues during an upgrade if the webhook is not responding.  Cluster administrators may want to disable the Knative Webhook on system namespaces to prevent issues during upgrades.

Reviewer: ```suggestion
The Knative webhook examines resources that are created, read, updated, or deleted. This includes system namespaces, which can cause issues during an upgrade if the webhook becomes non-responsive. Cluster administrators may want to disable the Knative webhook on system namespaces to prevent issues during upgrades.
```

Reviewer: This should be the intro :slightly_smiling_face: move to the top of the doc
